src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -118,10 +132,15 @@ protected void doWriteTo(StreamOutput out) throws IOException {
     protected void doXContent(XContentBuilder xContentBuilder, Params params) throws IOException {
         xContentBuilder.startObject(NAME);
         xContentBuilder.startObject(fieldName);
-        xContentBuilder.field(QUERY_TEXT_FIELD.getPreferredName(), queryText);
+        if (Objects.nonNull(queryText)) {
+            xContentBuilder.field(QUERY_TEXT_FIELD.getPreferredName(), queryText);
+        }
         if (Objects.nonNull(modelId)) {
             xContentBuilder.field(MODEL_ID_FIELD.getPreferredName(), modelId);
         }
+        if (Objects.nonNull(queryTokensSupplier) && Objects.nonNull(queryTokensSupplier.get())) {
```

Comment:
just curious, why we need supplier for queryTokens?

Reply:
It's due to the async model inference in doRewrite

Reply:
The doRewrite will be called multiple times. At the first time we register the async model inference and set the supplier non-null (although the model inference is not finished). Before the async inference end, when doRewrite is called again, we know we already have a registered model inference and will skip it.
Otherwise we'll register multiple inference for one query